docs/widget.rst

```diff
@@ -283,24 +292,47 @@ constructing the form.  The default resource registry
 (:attr:`deform.widget.resource_registry`) does not contain resource
 mappings for your newly-created requirement.
 
-If the required resources are tightly couopled to the new widget, it may
+If the required resources are tightly coupled to the new widget, it may
 be easier to supply the direct links to the logical requirements in the form:
 
 .. code-block:: python
-   :linenos:
+    :linenos:
 
-   from deform.widget import Widget
+    from deform.widget import Widget
 
-   class MyWidget(Widget):
+    class MyWidget(Widget):
        requirements = ( {
-           'js': ''my:static/path/to/jquery.js',
-           'css': [
-                'my:static/path/to/jquery.css', 
-                'my:static:path/to/bootstrap.css'],
+           "js": "my:static/path/to/jquery.js",
+           "css": [
+                "my:static/path/to/jquery.css",
+                "my:static:path/to/bootstrap.css"],
         } )
 
-The supplied paths sre resolved by ``request.get_path()`` so the requried
-static resources should be included in the pyramid config.
+The supplied paths are resolved by ``request.get_path()`` so the required
+static resources should be included in the Pyramid config.
+
+
+.. _dict-widget-requirements:
+
+Using dicts for specifying widget requirements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requirements specified as a sequence of dicts should be in the form``({requirement_type: requirement_location(s)}, ...)``.
+The ``requirement_type`` key must be either ``js`` or ``css``.
+The ``requirement_location(s)`` value must be either a string or a list of strings.
+Each string must resolve to a concrete resource.
+
+.. code-block:: python
+    :linenos:
+
+    from deform.widget import Widget
+
+    class MyWidget(Widget):
+        requirements = (
+            {"js": "deform:static/tinymce/tinymce.min.js"},
+            {"css": "deform:static/tinymce/tinymce.min.css"},
+        )
```

Comment:
this seems like almost the same example as on line 303-309. Do we need both?

Reply:
It would be good to include a code example for both methods.

Nonetheless, this came as a surprise to me. I did not know that we could already specify `requirements` for a widget via a dict as shown in this example. Does this mean that the work you did is for naught?

Reply:
I added that example, there are now three examples (two tuples, one added by me, one by you). Think we only need two

Reply:
My mistake. I thought that example pre-existed, but looking back at [commit history excluding my commits](https://github.com/Pylons/deform/pull/486/files/c77522fc4d1d8681379b1fb8288e7363a6d4839d), I see now. I split the narrative docs section into an introduction, and one subsection each for two-tuples and dicts, but didn't realize this.

Fixed in 88b4978 by combining the bits in pieces to flow better. @tdamsma Would you please review again?

Reply:
Much better!